drivers/i2c/i2c_esp32.c

```diff
@@ -262,13 +261,13 @@ static int i2c_esp32_recover(const struct device *dev)
 	return 0;
 }
 
-static void IRAM_ATTR i2c_esp32_configure_timeout(const struct device *dev)
+static void IRAM_ATTR i2c_esp32_configure_timeout(const struct device *dev, uint32_t bitrate)
```

Comment:
It sounds a bit strange to configure bit-rate in a `configure_timeout` function. I would either embed this into `i2c_esp32_configure()` or create a new `i2c_esp32_configure_btirate()` function. How does it sound? I guess 1st option is a bit straighforward.

Reply:
fair point.
I now also moved the data mode part in a separate function, which is now only executed in the init, as it is set by the device tree and isn't changing during runtime.